pinot-common/src/main/java/org/apache/pinot/common/metrics/base/PinotMetric.java

```diff
@@ -0,0 +1,34 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.pinot.common.metrics.base;
+
+
+public interface PinotMetric {
```

Comment:
Another high-level comment here is that shall we considering move all the metrics related interfaces into the spi module and create plugins for those.

Reply:
cc: @chenboat @Jackie-Jiang

Reply:
Moved all the interfaces under pinot-spi module.

Reply:
Nice! I think the next thing we can do (probably not in this PR) is that we can create pinot-metrics plugins and move yammer stuff to yammer-metrics and load it during runtime.